File: uikit/src/main/java/io/agora/chat/uikit/chat/viewholder/EaseVideoViewHolder.java
package io.agora.chat.uikit.chat.viewholder;

import android.content.Intent;
import android.view.View;

import androidx.annotation.NonNull;

import io.agora.chat.ChatClient;
import io.agora.chat.ChatMessage;
import io.agora.chat.FileMessageBody;
import io.agora.chat.VideoMessageBody;
import io.agora.chat.uikit.activities.EaseShowVideoActivity;
import io.agora.chat.uikit.interfaces.MessageListItemClickListener;
import io.agora.util.EMLog;


public class EaseVideoViewHolder extends EaseChatRowViewHolder {
    private static final String TAG = EaseVideoViewHolder.class.getSimpleName();

    public EaseVideoViewHolder(@NonNull View itemView, MessageListItemClickListener itemClickListener) {
        super(itemView, itemClickListener);
    }

    @Override
    public void onBubbleClick(ChatMessage message) {
        super.onBubbleClick(message);
        VideoMessageBody videoBody = (VideoMessageBody) message.getBody();
        EMLog.d(TAG, "video view is on click");
        if(ChatClient.getInstance().getOptions().getAutodownloadThumbnail()) {

        }else{
            if(videoBody.thumbnailDownloadStatus() == FileMessageBody.EMDownloadStatus.DOWNLOADING ||
                    videoBody.thumbnailDownloadStatus() == FileMessageBody.EMDownloadStatus.PENDING ||
                    videoBody.thumbnailDownloadStatus() == FileMessageBody.EMDownloadStatus.FAILED){
                // retry download with click event of user
                ChatClient.getInstance().chatManager().downloadThumbnail(message);
                return;
            }
        }
        Intent intent = new Intent(getContext(), EaseShowVideoActivity.class);
        intent.putExtra("msg", message);
        if (message != null && message.direct() == ChatMessage.Direct.RECEIVE && !message.isAcked()
                && message.getChatType() == ChatMessage.ChatType.Chat) {
            try {
                ChatClient.getInstance().chatManager().ackMessageRead(message.getFrom(), message.getMsgId());
            } catch (Exception e) {
                e.printStackTrace();
            }
        }
        getContext().startActivity(intent);
    }
}

File: ease-im-kit/src/main/kotlin/io/agora/uikit/feature/chat/viewholders/EaseVideoViewHolder.kt
package io.agora.uikit.feature.chat.viewholders

import android.view.View
import io.agora.uikit.feature.chat.activities.EaseShowVideoActivity
import io.agora.uikit.common.ChatClient
import io.agora.uikit.common.ChatDownloadStatus
import io.agora.uikit.common.ChatLog
import io.agora.uikit.common.ChatMessage
import io.agora.uikit.common.ChatMessageDirection
import io.agora.uikit.common.ChatType
import io.agora.uikit.common.ChatVideoMessageBody

class EaseVideoViewHolder(itemView: View) : EaseChatRowViewHolder(itemView) {
    override fun onBubbleClick(message: ChatMessage?) {
        super.onBubbleClick(message)
        message?.run {
            if (body !is ChatVideoMessageBody) {
                ChatLog.e(TAG, "message body is not video type")
                return
            }
            if (ChatClient.getInstance().options.autodownloadThumbnail) {

            } else {
                val body: ChatVideoMessageBody = body as ChatVideoMessageBody
                if (body.thumbnailDownloadStatus() === ChatDownloadStatus.DOWNLOADING
                    || body.thumbnailDownloadStatus() === ChatDownloadStatus.PENDING
                    || body.thumbnailDownloadStatus() === ChatDownloadStatus.FAILED) {
                    // retry download with click event of user
                    ChatClient.getInstance().chatManager().downloadThumbnail(this)
                    return
                }
            }
            if (direct() === ChatMessageDirection.RECEIVE && !isAcked && chatType === ChatType.Chat) {
                try {
                    ChatClient.getInstance().chatManager()
                        .ackMessageRead(from, msgId)
                } catch (e: Exception) {
                    e.printStackTrace()
                }
            }
            EaseShowVideoActivity.actionStart(mContext, message)
        }

    }

    companion object {
        private val TAG = EaseVideoViewHolder::class.java.simpleName
    }
}